codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/awsjson/AWSJsonHttpResponseBindingErrorGenerator.kt

```diff
@@ -28,33 +68,48 @@ class AWSJsonHttpResponseBindingErrorGenerator : HttpResponseBindingErrorGenerat
-                writer.openBlock(
-                    "public static func makeError(httpResponse: \$N, decoder: \$D) async throws -> \$N {", "}",
-                    ClientRuntimeTypes.Http.HttpResponse,
-                    ClientRuntimeTypes.Serde.ResponseDecoder,
-                    SwiftTypes.Error
+            with(writer) {
+                addImport(AWSSwiftDependency.AWS_CLIENT_RUNTIME.target)
+                addImport(SwiftDependency.CLIENT_RUNTIME.target)
+
+                openBlock(
+                    "public enum \$L: \$N {",
+                    "}",
+                    operationErrorName,
+                    ClientRuntimeTypes.Http.HttpResponseErrorBinding
                 ) {
-                    writer.write(
-                        "let restJSONError = try await \$N(httpResponse: httpResponse)",
-                        AWSClientRuntimeTypes.RestJSON.RestJSONError
-                    )
-                    writer.write("let requestID = httpResponse.requestId")
-                    writer.openBlock("switch restJSONError.errorType {", "}") {
-                        val errorShapes = op.errors.map { ctx.model.expectShape(it) as StructureShape }.toSet().sorted()
-                        for (errorShape in errorShapes) {
-                            var errorShapeName = errorShape.errorShapeName(ctx.symbolProvider)
-                            var errorShapeType = ctx.symbolProvider.toSymbol(errorShape).name
-                            writer.write(
-                                "case \$S: return try await \$L(httpResponse: httpResponse, decoder: decoder, message: restJSONError.errorMessage, requestID: requestID)",
-                                errorShapeName,
-                                errorShapeType
+                    openBlock(
+                        "public static func makeError(httpResponse: \$N, decoder: \$D) async throws -> \$N {", "}",
+                        ClientRuntimeTypes.Http.HttpResponse,
+                        ClientRuntimeTypes.Serde.ResponseDecoder,
+                        SwiftTypes.Error
+                    ) {
+                        write(
+                            "let restJSONError = try await \$N(httpResponse: httpResponse)",
+                            AWSClientRuntimeTypes.RestJSON.RestJSONError
+                        )
+                        write("let requestID = httpResponse.requestId")
+
+                        write("let serviceError = try await ${ctx.symbolProvider.toSymbol(ctx.service).name}Types.makeServiceError(httpResponse, decoder, restJSONError, requestID)")
+                        write("if let error = serviceError { return error }")
+
+                        openBlock("switch restJSONError.errorType {", "}") {
+                            val errorShapes =
+                                op.errors.map { ctx.model.expectShape(it) as StructureShape }.toSet().sorted()
```

Comment:
nit: can you fix this spacing? it should be 

```
val errorShapes = op.errors
    .map { ctx.model.expectShape(it) as StructureShape }
    .toSet()
    .sorted()
```

Reply:
Changed as suggested.